src/clr-angular/forms/select/option.ts

```diff
@@ -3,29 +3,40 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import {Component, ElementRef, HostBinding, HostListener, Inject, Optional} from "@angular/core";
+import {Component, ElementRef, HostBinding, HostListener, Inject, Input, OnDestroy, Optional} from "@angular/core";
+import {Subscription} from "rxjs";
+
 import {POPOVER_HOST_ANCHOR} from "../../popover/common/popover-host-anchor.token";
 import {IfOpenService} from "../../utils/conditional/if-open.service";
+
 import {OptionSelectionService} from "./providers/option-selection.service";
 
 @Component({selector: "clr-option", templateUrl: "option.html", host: {"[class.clr-option]": "true"}})
-export class ClrOption {
-    private _selected: boolean = false;
+export class ClrOption<T> implements OnDestroy {
+    private subscription: Subscription;
 
-    @HostBinding("class.active")
-    get selected(): boolean {
-        return this._selected;
-    }
+    @HostBinding("class.active") selected: boolean = false;
 
-    set selected(value: boolean) {
-        this._selected = value;
-    }
+    @Input("clrValue") value: T;
 
     constructor(private ifOpenService: IfOpenService, @Optional() @Inject(POPOVER_HOST_ANCHOR) parentHost: ElementRef,
-                public elRef: ElementRef, private optionSelectionService: OptionSelectionService) {
+                public elRef: ElementRef, private optionSelectionService: OptionSelectionService<T>) {
         if (!parentHost) {
             throw new Error("clr-option should only be used inside of a clr-select");
         }
+        this.initializeSubscription();
+    }
+
+    private initializeSubscription(): void {
+        this.subscription = this.optionSelectionService.valueChanged.subscribe((value: T) => {
+            if (value === null || value === undefined) {
```

Comment:
I should have added a comment here but if I don't do the null and undefined check and if the users do not assign any value to the options, all options get highlighted.

Reply:
added a comment